code/game/objects/objs.dm

```diff
@@ -227,3 +227,11 @@
 		current_skin = choice
 		icon_state = unique_reskin[choice]
 		to_chat(M, "[src] is now skinned as '[choice].'")
+
+/obj/proc/check_interactivity(mob/user)
+	var/turf/T = get_turf(src)
+	var/turf/P = get_turf(user)
+	if((P.Adjacent(T)) && user.IsAdvancedToolUser())

```

Comment:
You should use Adjacent directly not through turfs.

Reply:
What's not working correctly ? This whole proc seems bit redundant since canUseTopic exists already.

Reply:
interaction between mob and assembly components